### PR TITLE
[nfc] asan CI build

### PR DESCRIPTION
**Pull request by mikea**
I plan to add ubsan to the matrix next

### CODE REVIEW

File: .github/workflows/quick-test.yml [hunk 1]
```diff
@@ -46,12 +46,32 @@ jobs:
...
+    strategy:
+      fail-fast: false
+      matrix:
+        clang: [18]
+        config: [asan]
+    steps:
+      - uses: actions/checkout@v4
+      - name: install dependencies
+        # I observed 404s for some packages and added an `apt-get update`.
+        run: |
```
Comment (mikea) [inline .github/workflows/quick-test.yml:66]: not sure if there's a way to dedup this step here? @fhanau

Comment (fhanau) [reply .github/workflows/quick-test.yml:66]: You could add `driver: bazel-asan` `run-test:  cd c++ && bazel test --verbose_failures --test_output=errors -k --config=asan //...` to `Linux-clang` above and then filter out clang versions other than 18. Might not be much shorter though

Comment (mikea) [reply .github/workflows/quick-test.yml]: So, I wanted to try yaml anchors, but apparently they are not supported on github actions: https://github.com/orgs/community/discussions/4501

Comment (mikea) [reply .github/workflows/quick-test.yml]: OK, I did some bash hackery. How about now? @fhanau

**Review by fhanau — APPROVED** [review]

**Comment by fhanau** [comment]
LGTM, thought I had already approved yesterday. `libc++abi-$clang-dev` looks a bit wonky, but looks like it is working correctly and not trying to resolve `$clang-dev`.

**Pull request merged**